src/app/frontend/common/components/middleellipsis/middleellipsis.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @param {!Element} container
+ * @param {!Element} element
+ * @param {!Function} filter - middle ellipsis filter function used to truncate string
+ *
+ * @return {number}
+ */
+export default function computeTextLength(container, element, filter) {
```

Comment:
I will write this in documentation.

Element is the actual element which contains the text. It is inner element of container.